pkg/image/importer/importer.go

```diff
@@ -299,9 +319,35 @@ func formatRepositoryError(repository *importRepository, refName string, refID s
 	return
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image, cache *imageStreamImporterCache) error {
+	bs := repo.Blobs(ctx)
+
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+
+		if layerSize, ok := cache.LayerSizeCache[layer.Name]; ok {
+			size += layerSize
+			continue
+		}
+
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
+		if err != nil {
+			return err
+		}
+
+		cache.LayerSizeCache[layer.Name] = desc.Size
+		layer.LayerSize = desc.Size
+		size += desc.Size
```

Comment:
It is done in line [329](https://github.com/openshift/origin/pull/9115#discussion-diff-69731090R329)